src/server/express/http_rest_handler.ts

```diff
@@ -0,0 +1,471 @@
+  // ============================================================================
+
+  /**
+   * GET /v1/card
+   *
+   * Retrieves the authenticated extended agent card.
+   *
+   * @returns 200 OK with agent card
+   * @returns 500 Internal Server Error on failure
+   */
+  router.get(
+    '/v1/card',
+    asyncHandler(async (req, res) => {
+      const context = await buildContext(req);
+      const result = await restTransportHandler.getAuthenticatedExtendedAgentCard();
+      sendResponse(res, HTTP_STATUS.OK, result, context);
+    })
+  );
+
+  /**
+   * POST /v1/message:send
+   *
+   * Sends a message to the agent synchronously.
+   * Returns either a Message (for immediate responses) or a Task (for async processing).
+   * Note: Colon is escaped in route definition for Express compatibility.
+   *
+   * @param req.body - RestMessageSendParams (snake_case format)
+   * @returns 201 Created with RestMessage or RestTask
+   * @returns 400 Bad Request if message is invalid
+   */
+  router.post(
+    '/v1/message\\:send',
+    asyncHandler(async (req, res) => {
+      const context = await buildContext(req);
+      const result = await restTransportHandler.sendMessage(
+        req.body as RestMessageSendParams,
+        context
+      );
+      sendResponse(res, HTTP_STATUS.CREATED, result, context);
+    })
+  );
+
+  /**
+   * POST /v1/message:stream
+   *
+   * Sends a message to the agent with streaming response.
+   * Returns a Server-Sent Events (SSE) stream of updates.
+   * Note: Colon is escaped in route definition for Express compatibility.
+   *
+   * @param req.body - RestMessageSendParams (snake_case format)
+   * @returns 200 OK with SSE stream of messages, tasks, and status updates
+   * @returns 400 Bad Request if message is invalid
+   * @returns 501 Not Implemented if streaming not supported
+   */
+  router.post(
+    '/v1/message\\:stream',
+    asyncHandler(async (req, res) => {
+      const context = await buildContext(req);
+      const stream = await restTransportHandler.sendMessageStream(
+        req.body as RestMessageSendParams,
+        context
+      );
+      await sendStreamResponse(res, stream, context);
+    })
+  );
+
+  /**
+   * GET /v1/tasks/:taskId
+   *
+   * Retrieves the current status and details of a task.
+   *
+   * @param req.params.taskId - Task identifier
+   * @param req.query.historyLength - Optional number of history messages to include
+   * @returns 200 OK with RestTask
+   * @returns 400 Bad Request if historyLength is invalid
+   * @returns 404 Not Found if task doesn't exist
+   */
+  router.get(
+    '/v1/tasks/:taskId',
+    asyncHandler(async (req, res) => {
+      const context = await buildContext(req);
+      const result = await restTransportHandler.getTask(
+        req.params.taskId,
+        req.query.historyLength,
+        context
+      );
+      sendResponse(res, HTTP_STATUS.OK, result, context);
+    })
+  );
+
+  /**
+   * POST /v1/tasks/:taskId\\:cancel
```

Comment:
nit: to avoid copying escaping characters to docs we can use [this notation](https://a2a-protocol.org/latest/specification/#53-method-mapping-reference) in comments, so here it'd be
```suggestion
   * POST /v1/tasks/{id}:cancel
```

(here and below)

Reply:
thanks! it's fixed